### PR TITLE
pull request for Issue 65

### PR DESCRIPTION
**Pull request by NickPolyder**
Hello here is the pull request for issue #65.

Please have a look and lets talk if there is a need for change

Thanks

### CODE REVIEW

File: MainProgram/AlgorithmsTests/BinaryTreeRecursiveWalkerTests.cs [hunk 1]
```diff
@@ -1,185 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/AlgorithmsTests/BinaryTreeRecursiveWalkerTests.cs]: Why did you delete this file?

File: MainProgram/AlgorithmsTests/GraphsBreadthFirstSearchTest.cs [hunk 2]
```diff
@@ -1,72 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/AlgorithmsTests/GraphsBreadthFirstSearchTest.cs]: Why did you delete this file?

File: MainProgram/AlgorithmsTests/HeapSorterTest.cs [hunk 3]
```diff
@@ -1,31 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/AlgorithmsTests/HeapSorterTest.cs]: Why did you delete this file?

File: MainProgram/AlgorithmsTests/InsertionSortTest.cs [hunk 4]
```diff
@@ -1,56 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/AlgorithmsTests/InsertionSortTest.cs]: Why did you delete this file?

File: MainProgram/AlgorithmsTests/MergeSorterTest.cs [hunk 5]
```diff
@@ -1,21 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/AlgorithmsTests/MergeSorterTest.cs]: Why did you delete this file?

File: MainProgram/AlgorithmsTests/QuickSortTest.cs [hunk 6]
```diff
@@ -1,18 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/AlgorithmsTests/QuickSortTest.cs]: Why did you delete this file?

File: MainProgram/DataStructuresTests/ArrayListTest.cs [hunk 7]
```diff
@@ -1,84 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/DataStructuresTests/ArrayListTest.cs]: Why did you delete this file?

File: MainProgram/DataStructuresTests/BinaryHeapsTest.cs [hunk 8]
```diff
@@ -1,44 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/DataStructuresTests/BinaryHeapsTest.cs]: Why did you delete this file?

File: DataStructures/Dictionaries/OpenAddressingHashTable.cs [hunk 9]
```diff
@@ -0,0 +1,395 @@
+﻿using System;
```
Comment (aalhour) [inline DataStructures/Dictionaries/OpenAddressingHashTable.cs]: This is a new data structure implementation, correct? Can you maybe take this out of this PR and submit it as a separate one, this PR is kind of bloated, it has fixes for the warnings as well as many other things, making this new contribution to the code kind of lost in a huge PR.

File: DataStructures/Trees/AugmentedBinarySearchTree.cs [hunk 10]
```diff
@@ -146,7 +143,7 @@ protected bool _remove(BSTRankedNode<T> node)
 
             if (node.ChildrenCount == 2) // if both children are present
             {
-                var successor = node.RightChild;
+                var successor = _findNextLarger(node);
```
Comment (aalhour) [inline DataStructures/Trees/AugmentedBinarySearchTree.cs]: Can you tell me why did you change this line?

File: MainProgram/AlgorithmsTests/SieveOfEratosthenesTest.cs [hunk 11]
```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
```
Comment (aalhour) [inline MainProgram/AlgorithmsTests/SieveOfEratosthenesTest.cs]: All unit tests should be added to the `UnitTest` project. The `MainProgram` project is deprecated and will be removed soon.

File: MainProgram/DataStructuresTests/DLinkedListTest.cs [hunk 12]
```diff
@@ -1,150 +0,0 @@
-﻿using System;
```
Comment (aalhour) [inline MainProgram/DataStructuresTests/DLinkedListTest.cs]: Why did you delete this file?

File: MainProgram/DataStructuresTests/OpenAddressingHashTableTest.cs [hunk 13]
```diff
@@ -0,0 +1,235 @@
+﻿using System;
```
Comment (aalhour) [inline MainProgram/DataStructuresTests/OpenAddressingHashTableTest.cs]: All new unit tests should be added to the `UnitTest` project. The `MainProgram` project is deprecated and will be removed soon. Please move this to the `UnitTest` project.

**Comment by aalhour** [comment]
Hello @NickPolyder,

Thanks for your PR. The PR is huge and contains many changes, it deletes some source files from the `MainProgram`, it adds two new components: Open Addressing Hash Table and the Sieve of Eratosthenes, other than introducing some clean ups.

It would be great if you can take out the new additions - the hash table and the sieve algorithm - and submit them as two separate PRs from this one, and keep the scope of this PR focused on cleanups.

**Comment by aalhour** [comment]
Hello @ivandrofly,

I have added you as a collaborator on the project. It would be great if you can give me a helping hand in reviewing this PR. I have added a short summary of the notes in my last comment, feel free to go through it with fresh eyes and submit your comments. The PR is packed.

Cheers and thanks,
Ahmad

**Comment by NickPolyder** [comment]
Hello @aalhour  and @ivandrofly. My PR is about this commit:
 https://github.com/NickPolyder/C-Sharp-Algorithms/commit/ebaa35bfbc806ce29290a9b60483169760e64091 
I do not know why and how all this information that you commented is related to this PR.
